src/lib/APS2.cpp

```diff
@@ -178,82 +211,90 @@ double APS2::get_fpga_temperature(){
-	FILE_LOG(logDEBUG1) << "Read " << fileData.size() << " bytes in bitfile.";
-
-	//Pad out to align with a multiple of 8 bytes
-	//This is because DRAM writes must be mulitples of 8 bytes and 8 bytes aligned
-	int padBytes = (8 - (fileData.size() % 8)) % 8;
-	FILE_LOG(logDEBUG1) << "Padding bitfile byte vector with " << padBytes << " bytes.";
-	fileData.resize(fileData.size() + padBytes, 0xff);
-
-	//Copy over the file data to a 32 bit data vector
-	vector<uint32_t> packedData;
-	packedData.resize(fileData.size()/4);
-	memcpy(packedData.data(), fileData.data(), fileData.size());
-
-	//Convert to big endian byte order - basically because it will be byte-swapped again when the packet is serialized
-	for (auto & packet : packedData) {
-		packet = htonl(packet);
+	//Get the file size in bytes
+	std::ifstream FID (bitFile, std::ios::in|std::ios::binary);
+  if (!FID.is_open()){
+		FILE_LOG(logERROR) << "Unable to open bitfile: " << bitFile;
+    throw APS2_NO_SUCH_BITFILE;
+  }
+
+  FID.seekg(0, std::ios::end);
+  size_t file_size = FID.tellg();
+	FILE_LOG(logDEBUG) << "Opened bitfile: " << bitFile << " with " << file_size << " bytes";
+  FID.seekg(0, std::ios::beg);
+
+	//Figure out padding size for alignment
+	size_t padding_bytes = (alignment - (file_size % alignment)) % alignment;
+	FILE_LOG(logDEBUG1) << "Padding bitfile byte vector with " << padding_bytes << " bytes.";
+  //Copy the file data to a 32bit word vector
+  vector<uint32_t> bitfile_words((file_size+padding_bytes)/4, 0xffffffff);
+  FID.read(reinterpret_cast<char *>(bitfile_words.data()), file_size);
+
+	//Swap bytes because we want to preserve bytes order in file it will be byte-swapped again when the packet is serialized
+	for (auto & val : bitfile_words) {
+		val = htonl(val);
 	}
 
-	FILE_LOG(logDEBUG1) << "Bit file is " << packedData.size() << " 32-bit words long";
-
-	uint32_t addr = 0; // TODO: make start address depend on position
-	auto packets = pack_data(addr, packedData, APS_COMMANDS::FPGACONFIG_ACK);
-
-	// send in groups of 20
-	ethernetRM_->send(deviceSerial_, packets, 20);
-}
-
-int APS2::select_image(const int & bitFileNum) {
-	FILE_LOG(logINFO) << "Selecting bitfile number " << bitFileNum;
-
-	uint32_t addr = 0; // todo: make start address depend on bitFileNum
-
-	APSEthernetPacket packet;
-	packet.header.command.r_w = 0;
-	packet.header.command.cmd =  static_cast<uint32_t>(APS_COMMANDS::FPGACONFIG_CTRL);
-	packet.header.command.cnt = 0;
-	packet.header.addr = addr;
+	switch (media) {
+		case BITFILE_MEDIA_DRAM:
+			write_configuration_SDRAM(start_addr, bitfile_words);
+			break;
+		case BITFILE_MEDIA_EPROM:
+			write_flash(start_addr, bitfile_words);
+			break;
+	}
 
-	return ethernetRM_->send(deviceSerial_, packet, false);
+	//Now validate bitfile data (EPROM only for now)
+	//TODO: DRAM as well?
+	flash_validate_done = 0;
+	if (media == BITFILE_MEDIA_EPROM) {
+		uint32_t end_addr = start_addr + 4*bitfile_words.size();
+		uint32_t addr{start_addr};
```

Comment:
I find this style of variable initialization to be less clear than `uint32_t addr = start_addr`.